Serviceoffloading/knox handling

Chikdolman opened the pull request:
Adds Mouse, Keyboard input for Knox enabled devices.

Chikdolman committed:
[Service Offloading] Key/Mouse Input migration
Adds patches for keyboard/Mouse Event on Android.
It only works when Knox SDK is enabled on Galaxy.

Chikdolman committed:
[Service offloading] Merge for conflicts
Fix conflicts

Sunwoo-Nam commented on chrome/android/BUILD.gn:
```diff
@@ -72,7 +72,11 @@ chrome_jni_registration_header =
 jinja_template("chrome_public_android_manifest") {
-  input = "java/AndroidManifest.xml"
+  if (enable_service_offloading_knox) {
+    input = "java/AndroidManifest_service_offloading_knox.xml"
```
I think we can use AndroidManifest.xml for Knox API.

Chikdolman replied:
THank you. Done

Sunwoo-Nam commented on content/browser/android/content_startup_flags.cc:
```diff
@@ -87,6 +87,12 @@ void SetContentCommandLineFlags(bool single_process) {
+  // renderer process. Currently, the Knox permission only applies to the
+  // browser process and it cannot be propagated to the renderer process.
+  // Therefore we apply single process mode for Knox system input.
+  parsed_command_line->AppendSwitch(switches::kSingleProcess);
```
The single process mode should be enabled if |enable_service_offloading_knox| is true.

Chikdolman replied:
Thank you. Done

Sunwoo-Nam commented on third_party/blink/renderer/modules/modules_idl_files.gni:
```diff
@@ -528,13 +528,16 @@ if (!is_android) {
-  modules_idl_files += get_path_info(["input_control/input_control.idl"], "abspath")
+
+if (is_win || is_android) {
+  if(enable_service_offloading){
```
Why do you change the condition?

Chikdolman replied:
Rolled back

Sunwoo-Nam reviewed (commented):
Please remove the commit for conflict.

Chikdolman committed:
[Service Offloading] Supports input handling
Support Key event and mouse event
For Android, only Galaxy phone is enabled with the patch.
Even for Galaxy, the Knox library should be manually added
